cleanup/artifactCleanup/ArtifactCleanupTest.groovy

```diff
@@ -1,41 +1,112 @@
 
         def builder = artifactory.repositories().builders()
-        def repository = builder.localRepositoryBuilder().key('maven-local')
-        .repositorySettings(new MavenRepositorySettingsImpl()).build()
+        def repository = builder.localRepositoryBuilder().key(repoName)
+            .repositorySettings(new MavenRepositorySettingsImpl()).build()
         artifactory.repositories().create(0, repository)
 
+        return artifactory
+    }
+
+    def 'artifact cleanup test'() {
+        setup:
+        def artifactory = createClientAndMavenRepo('maven-local')
+
         def file = new ByteArrayInputStream('test'.bytes)
-        artifactory.repository("maven-local").upload('test', file).doUpload()
+        artifactory.repository('maven-local').upload('test', file).doUpload()
 
         when:
         artifactory.plugins().execute('cleanup').
             withParameter('repos', 'maven-local').sync()
-        artifactory.repository("maven-local").file('test').info()
+        artifactory.repository('maven-local').file('test').info()
+
+        then:
+        notThrown(HttpResponseException)
+
+        when:
+        artifactory.plugins().execute('cleanup').
+            withParameter('repos', 'maven-local').
+            withParameter('months', '0').sync()
+        artifactory.repository('maven-local').file('test').info()
+
+        then:
+        thrown(HttpResponseException)
+
+        cleanup:
+        artifactory.repository('maven-local').delete()
+    }
+
+    def 'artifact cleanup skip artifact test'() {
+        setup:
+        def artifactory = createClientAndMavenRepo('maven-local')
+
+        def file = new ByteArrayInputStream('test'.bytes)
+        artifactory.repository('maven-local').upload('test', file).withProperty('cleanup.skip', 'true').doUpload()
+
+        when:
+        artifactory.plugins().execute('cleanup').
+            withParameter('repos', 'maven-local').
+            withParameter('months', '0').sync()
+        artifactory.repository('maven-local').file('test').info()
 
         then:
         notThrown(HttpResponseException)
 
         when:
         artifactory.plugins().execute('cleanup').
             withParameter('repos', 'maven-local').
+            withParameter('months', '0').
+            withParameter('disablePropertiesSupport', 'true').sync()
+        artifactory.repository('maven-local').file('test').info()
+
+        then:
+        thrown(HttpResponseException)
+
+        cleanup:
+        artifactory.repository('maven-local').delete()
+    }
+
+    def 'artifact cleanup skip folder test'() {
+        setup:
+        def artifactory = createClientAndMavenRepo('maven-local')
+
+        def file = new ByteArrayInputStream('test'.bytes)
+        artifactory.repository('maven-local').upload('foo/bar/test', file).withProperty('cleanup.skip', 'true').doUpload()
+        artifactory.repository('maven-local').folder('foo').properties().addProperty('cleanup.skip', 'true').doSet();
```

Comment:
@elioengcomp : Thanks for the feedback. This test was more to test the skip paths prefixes calculation (this [part](https://github.com/JFrogDev/artifactory-user-plugins/pull/122/files#diff-9f26e4f8e3ac0a66460262a10b008373R181) of code). But you are right, I will add a `when`/`then` case for each levels.

> What is the behavior when we set the property at the artifact's parent folder only? 
> What about when we set the property at the parent's parent folder only?

The childrens of path will be skipped in the cleanup (implicit recursion, due to [RTFACT-14738](https://www.jfrog.com/jira/browse/RTFACT-14738)).
I can complete the documentation if the understood is not trivial.